File: src/main/java/com/big/dreamer/doccentral/document/carsale/model/DocumentDetails.java
package com.big.dreamer.doccentral.document.carsale.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotBlank;

public record DocumentDetails(
        @JsonProperty("calidad_de") @NotBlank String garment,
        @JsonProperty("institucion") String institution,
        @JsonProperty("precio") @NotBlank String price,
        @JsonProperty("domicilio") @NotBlank String settlement,
        @JsonProperty("departamento") @NotBlank String state,
        @JsonProperty("fecha_firma") @NotBlank String signDate,
        @JsonProperty("hora_firma") @NotBlank String signHour,
        @JsonProperty("identifica_vendedor") @NotBlank String identifiesSeller,
        @JsonProperty("identifica_comprador") @NotBlank String identifiesBuyer) {
}

File: src/main/kotlin/com/big/dreamer/doccentral/car/sale/dto/DocumentDetails.kt
package com.big.dreamer.doccentral.car.sale.dto

import com.fasterxml.jackson.annotation.JsonAlias

data class DocumentDetails(
    @JsonAlias("calidad_de")
    val garment: String = "",
    @JsonAlias("institucion")
    var institution: String = "",
    @JsonAlias("precio")
    val price: String = "",
    @JsonAlias("domicilio")
    val settlement: String = "",
    @JsonAlias("departamento")
    val state: String = "",
    @JsonAlias("fecha_firma")
    val signDate: String = "",
    @JsonAlias("hora_firma")
    val signHour: String = "",
    @JsonAlias("identifica_vendedor")
    var identifiesSeller: String = "No",
    @JsonAlias("identifica_comprador")
    var identifiesBuyer: String = "No",
)